monai/transforms/compose.py

```diff
@@ -308,54 +369,44 @@ class Compose(Randomizable, InvertibleTransform):
-            for other metadata, log the values directly. default to `False`.
-        lazy_evaluation: whether to enable lazy evaluation for lazy transforms. If False, transforms will be
-            carried out on a transform by transform basis. If True, all lazy transforms will
-            be executed by accumulating changes and resampling as few times as possible.
-            A `monai.transforms.Identity[D]` transform in the pipeline will trigger the evaluation of
+        lazy: whether to enable lazy evaluation for lazy transforms. This is an optional bool that can take
+            the following values. If lazy=False, lazy execution is disabled and transforms will be
+            carried out on a transform by transform basis. If lazy=True, all lazy transforms will
+            be executed by accumulating changes and resampling as few times as possible. If lazy is None,
+            Compose will perform lazy execution on lazy transforms that have their lazy flag set to True.
+            A `monai.transforms.ApplyPending[d]` transform in the pipeline will trigger the evaluation of
             the pending operations and make the primary data up-to-date.
         overrides: this optional parameter allows you to specify a dictionary of parameters that should be overridden
             when executing a pipeline. These each parameter that is compatible with a given transform is then applied
-            to that transform before it is executed. Note that overrides are currently only applied when lazy_evaluation
-            is True. If lazy_evaluation is False they are ignored.
+            to that transform before it is executed. Note that overrides are currently only applied when lazy
+            is True. If lazy is False they are ignored.
             currently supported args are:
             {``"mode"``, ``"padding_mode"``, ``"dtype"``, ``"align_corners"``, ``"resample_mode"``, ``device``},
             please see also :py:func:`monai.transforms.lazy.apply_transforms` for more details.
-        override_keys: this optional parameter specifies the keys to which ``overrides`` are to be applied. If
-            ``overrides`` is set, ``override_keys`` must also be set.
-        verbose: whether to print debugging info when lazy_evaluation=True.
+        logger_name: this optional parameter allows you to specify a logger by name. If this is not set
+            it defaults to 'Compose'. You can also suppress logging by setting this to None.
     """
 
     def __init__(
         self,
         transforms: Sequence[Callable] | Callable | None = None,
         map_items: bool = True,
         unpack_items: bool = False,
-        log_stats: bool = False,
-        lazy_evaluation: bool | None = None,
+        lazy: bool | None = False,
         overrides: dict | None = None,
-        override_keys: Sequence[str] | None = None,
-        verbose: bool = False,
+        options: dict | None = None,
```

Comment:
please add documentation for this `options` then I guess we can start reviewing the PR?